tsdb/db.go

```diff
@@ -749,6 +749,33 @@ func open(dir string, l log.Logger, r prometheus.Registerer, opts *Options, rngs
 
 	if initErr := db.head.Init(minValidTime); initErr != nil {
 		db.head.metrics.walCorruptionsTotal.Inc()
+		if _, isCheckpointCorruption := initErr.(*wal.CorruptionErr); isCheckpointCorruption && wlog != nil {
```

Comment:
This error could also mean just the WAL corruption and not the checkpoint corruption. Init() could returning some fixed `ErrCheckpointCorrupted` to identify checkpoint corruption.

Reply:
Yes, that was in my code for next commit which is now pushed 😄